app/src/main/java/app/insti/fragment/ComplaintDetailsFragment.java

```diff
@@ -0,0 +1,242 @@
+import android.support.v4.app.Fragment;
+import android.support.v4.view.ViewPager;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.Toast;
+
+import java.util.Objects;
+
+import app.insti.R;
+import app.insti.activity.MainActivity;
+import app.insti.adapter.ComplaintDetailsPagerAdapter;
+import app.insti.adapter.ImageViewPagerAdapter;
+import app.insti.api.RetrofitInterface;
+import app.insti.api.model.User;
+import app.insti.api.model.Venter;
+import me.relex.circleindicator.CircleIndicator;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class ComplaintDetailsFragment extends Fragment {
+
+    private static final String TAG = ComplaintDetailsFragment.class.getSimpleName();
+    Button buttonVoteUp;
+    TabLayout slidingTabLayout;
+    ViewPager viewPager;
+    View mview;
+    private String complaintId, sessionID, userId;
+    private ComplaintDetailsPagerAdapter complaintDetailsPagerAdapter;
+    private int voteCount;
+    CircleIndicator circleIndicator;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_complaint_details, container, false);
+
+        LinearLayout imageViewHolder = view.findViewById(R.id.image_holder_view);
+        CollapsingToolbarLayout.LayoutParams layoutParams = new CollapsingToolbarLayout.LayoutParams
+                (CollapsingToolbarLayout.LayoutParams.MATCH_PARENT,
+                        getResources().getDisplayMetrics().heightPixels / 2);
+        imageViewHolder.setLayoutParams(layoutParams);
+
+        buttonVoteUp = view.findViewById(R.id.buttonVoteUp);
+        slidingTabLayout = view.findViewById(R.id.sliding_tab_layout);
+        circleIndicator = view.findViewById(R.id.indicator);
+        this.mview = view;
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        buttonVoteUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                upVote();
+            }
+        });
+
+        Bundle bundle = getArguments();
+        complaintId = bundle.getString("id");
+        sessionID = bundle.getString("sessionId");
+        userId = bundle.getString("userId");
+
+        if (bundle != null) {
+            Log.i(TAG, "bundle != null");
+            callServerToGetDetailedComplaint();
+        }
+    }
+
+    private void upVote() {
+        if (voteCount == 0) {
+            RetrofitInterface retrofitInterface = ((MainActivity) getActivity()).getRetrofitInterface();
```

Comment:
MainActivity$getRetrofitInterface will not exist after #218

Reply:
Again, tell me the solution

Reply:
`Utils$getRetrofitInterface`